src/compaction/mod.rs

```diff
@@ -329,8 +330,7 @@ where
             }
             max = Some(key.value.to_key());
 
-            // FIXIT: it is not the real size in arrows;
-            written_size += key.size();
+            written_size += key.size() + entry.value().size();
```

Comment:
I think it should use the arrow array's buffer size here, see: https://docs.rs/arrow/latest/arrow/?search=values_slice